IPs/RSKIP502.md

```diff
@@ -0,0 +1,252 @@
+```
+
+**Response codes:**
+
+- 0: Success – The locking cap has been adjusted successfully.
+- -1: Unauthorized caller – Only authorized entities can adjust the locking cap.
+- -2: Invalid value – The specified cap value is invalid (e.g., less than the current cap or excessive).
+- -10: Generic error – A non-specific error occurred while processing the request.
+
+### 5. Method for Transferring RBTC to the Union Bridge
+
+A new method will be added to the Bridge contract to allow the transfer of RBTC to the Union Bridge. The method will have the following restrictions:
+
+- Only callable by the Union Bridge contract address.
+- The total requested amount (current request + previously requested amounts) must not exceed the current locking cap.
+
+**Method Signature:** 
+
+```
+function requestUnionRBTC(uint256 amountInWeis) public returns (int256)
+```
+
+**Response codes:**
+
+- 0: Success – Funds were successfully transferred to the Union Bridge contract, and the internal tracking was updated accordingly.
+- -1: Unauthorized caller – The caller is not the Union Bridge contract address.
+- -2: Invalid value – The amount requested, combined with previously requested amounts, exceeds the current locking cap value.
+- -10: Generic error – A non-specific error occurred while processing the request.
+
+**Event**
+
+An event will be emitted each time this method is invoked, to provide transparency about the transfer.
+
+```
+event union_rbtc_requested(address indexed requester, uint256 amountInWeis);
+```
+
+### 6. Tracking transfers to the Union Bridge
+
+A new storage entry will be used to track the total amount of RBTC that has been transferred to the Union Bridge. This is essential to ensure that the locking cap is not exceeded.
+
+**Storage entry name:** weisTransferredToUnionBridge
+
+Tracks the total amount of RBTC (expressed in weis) transferred to the Union Bridge.
+
+### 7. Receiving funds back from the Union Bridge
+
+The Union Bridge contract will have the capability to send funds back to the PowPeg. When this happens, the tracking entry for the amount transferred will need to be updated to reflect the returned RBTC. The Bridge will not allow **weisTransferredToUnionBridge** to go negative, which could only happen if the Union Bridge is malfunctioning. If this happens, the method will fail with an error code and the Union-PowPeg integration will automatically enter pause mode.
+
+A new method will be added to the Bridge contract to allow the transfer of RBTC from the Union Bridge back to the PowPeg. The method will have the following restrictions:
+
+- It will only be callable by the Union Bridge contract address.
+- The amount sent back cannot exceed the total amount of RBTC previously transferred to the Union Bridge, as tracked by the **weisTransferredToUnionBridge** storage entry. In other words, the Union Bridge cannot return more RBTC than it has previously received.
```

Comment:
Since we are dealing with RBTC should the value be measured in sats rather than weis?